docs/DeveloperGuide.md

```diff
@@ -1,14 +1,123 @@
+  - [Update feature](#Update-feature)
+  - [EquipmentManager](#EquipmentManager)
+- [Product scope](#Product scope)
+  - [Target user profile](#Target-user-profile)
+  - [Value proposition](#Value-proposition)
+- [User stories](#User-Stories)
+- [NonFunctional requirements](#NonFunctional Requirements)
+- [Glossary](#Glossary)
+- [Instructions for manual testing](#Instructions-for-manual-testing)
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+Our project could not have been possible without the prior work of the following:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+- [SE-EDU project team](https://se-education.org/docs/team.html) for their work on [AddressBook Level 3](https://github.com/se-edu/addressbook-level3), for which we referenced ideas as well as code snippets.
+- Google's [GSON Java library](https://github.com/google/gson) for which we used to load and save our data files.
+- Members of the [AY2122S2-CS2113-F10-2 team](https://github.com/nus-cs2113-AY2122S2) for pointing out the possibility of and requesting permission to make use of the above GSON library.  
 
+--------------------------------------------------------------------------------------------------------------------
+## **Design and Implementation**
 
+This section describes some noteworthy details on how certain features are designed and implemented.
+
+### Architecture
+
+Our design draws significant inspiration from the implementation of AddressBook Level 3 (henceforth AddressBook). 
+As the program was initially conceptualised to be a text-based command-line interface, heavy consideration was given
+to the design and user interactions when it came to the text input to be parsed by the user. 
+As such, we referenced AddressBook to segment the program into **`Parser`**, **`Command`** and **`Equipment`** classes.
+
+**Main components of the architecture**
+
+- **`Duke`** is the Main class and entry-point for the program. 
+- **`Parser`** serves as the first filter to split raw text input and pass in arguments to the **`Command`** for further processing.
+- **`Command`**, together with its various subclasses serve as specific implementations to pass the arguments taken in to the various methods of the **`EquipmentManager`** class.
+- **`EquipmentManager`** keeps track of actual **`Equipment`** instances created by our program.
+- **`Storage`** performs File I/O functions.
+
+### Parser
+
+**`Parser`** is the first filter for text inputs read to the user. 
+As alluded to prior, one major consideration was to build it in a manner that can parse text input as effectively as possible. In considering text input, we divided the parsing into the following segments:
+
+#### `commandWord` `argumentTag/argumentValue` `argumentTag/argumentValue` `...`
```

Comment:
Perhaps there's some mistyping for the last one? It's the same as the second one.